frontend/src/components/core/Input.tsx

```diff
@@ -148,8 +148,8 @@ export const NumberInput = styled(Input).attrs(() => ({
   }
 `;
 
-export const TextInput = styled(Input).attrs(() => ({
-  type: 'text',
+export const TextInput = styled(Input).attrs((props) => ({
+  type: (props?.type ? props.type : 'text'),
 }))`
```

Comment:
Given that this component is named `TextInput`, not sure about being able to change its type - maybe rename the component or make a new one specific for that other use?

Reply:
Good point, this was mainly a workaround to get passwords to work, but I created a separate component for that as suggested. I also actually refactored the LoginInput and RegisterInput into a single AuthInput and AuthPasswordInput in the `Input.tsx` file, cause otherwise there'd be four almost duplicate components across the login and register files (LoginInput, LoginPasswordInput, RegisterInput, RegisterPasswordInput).

Reply:
Looks good, thanks! 👍